src/main/java/hudson/remoting/FastPipedInputStream.java

```diff
@@ -175,6 +175,7 @@ public int read(@Nonnull byte[] b, int off, int len) throws IOException {
                     try {
                         buffer.wait(FastPipedOutputStream.TIMEOUT);
                     } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
```

Comment:
This is certainly considered good practice, but it will require good testing before changing the interrupted behavior in areas like this.